src/cls/rbd/cls_rbd_types.cc

```diff
@@ -316,5 +317,138 @@ bool GroupSpec::is_valid() const {
   return (!group_id.empty()) && (pool_id != -1);
 }
 
+void GroupSnapshotNamespace::encode(bufferlist& bl) const {
+  ENCODE_START(1, 1, bl);
```

Comment:
Minor: probably no need to encode another version here since it's already wrapped in a versioned struct